forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,329 @@
+                list(&config, snapshot_dir);
+            }
+            Self::Remove {
+                filename,
+                snapshot_dir,
+                force,
+            } => {
+                remove(&config, filename, snapshot_dir, *force);
+            }
+            Self::Prune {
+                snapshot_dir,
+                force,
+            } => {
+                prune(&config, snapshot_dir, *force);
+            }
+        }
+    }
+}
+
+fn list(config: &Config, snapshot_dir: &Option<PathBuf>) {
+    let snapshot_dir = snapshot_dir
+        .clone()
+        .unwrap_or_else(|| default_snapshot_dir(config));
+    println!("Snapshot dir: {:?}", snapshot_dir);
+    if let Ok(dir) = fs::read_dir(snapshot_dir) {
+        println!("\nLocal snapshots:");
+        for path in dir
+            .flatten()
+            .map(|entry| entry.path())
+            .filter(|p| p.is_file())
+        {
+            if let Some(Some(filename)) = path.file_name().map(|n| n.to_str()) {
+                if filename.ends_with(".car") {
+                    println!("{filename}");
+                }
+            }
+        }
+    }
+}
+
+fn remove(config: &Config, filename: &PathBuf, snapshot_dir: &Option<PathBuf>, yes: bool) {
+    let snapshot_dir = snapshot_dir
+        .clone()
+        .unwrap_or_else(|| default_snapshot_dir(config));
+    let mut snapshot_path = snapshot_dir;
+    snapshot_path.push(filename);
+    if snapshot_path.exists()
+        && snapshot_path.is_file()
+        && snapshot_path.extension() == Some(OsStr::new("car"))
+    {
+        println!("Deleting {:?}", snapshot_path);
+        if !yes && !prompt_confirm() {
+            println!("Aborted.");
+            return;
+        }
+
+        delete_snapshot(&snapshot_path);
+    } else {
+        println!(
+                "{:?} is not a valid snapshot file path, to list all snapshots, run forest-cli snapshot list",
+                snapshot_path);
+    }
+}
+
+fn prune(config: &Config, snapshot_dir: &Option<PathBuf>, yes: bool) {
+    {
+        let snapshot_dir = snapshot_dir
+            .clone()
+            .unwrap_or_else(|| default_snapshot_dir(config));
+        println!("Snapshot dir: {:?}", snapshot_dir);
+        let mut snapshots_with_valid_name = vec![];
+        let mut snapshot_to_keep = None;
+        let mut latest_date = Date::MIN;
+        let mut latest_height = 0;
+        let pattern = Regex::new(
+            r"^forest_snapshot_([^_]+?)_(?P<date>\d{4}-\d{2}-\d{2})_height_(?P<height>\d+).car$",
+        )
+        .unwrap();
+        if let Ok(dir) = fs::read_dir(snapshot_dir) {
+            println!("\nLocal snapshots:");
+            for path in dir
+                .flatten()
+                .map(|entry| entry.path())
+                .filter(|p| p.is_file())
+            {
+                if let Some(Some(filename)) = path.file_name().map(|n| n.to_str()) {
+                    let captures = pattern.captures(filename);
```

Comment:
do we need to do so much `if-letting` here? If we match the regular expression, then the unwraps shouldn't be fallible, no?

Reply:
I thought @lemmih mentioned we want to remove all `expect`, `unwrap` usage someday, I can do `unwrap` if that's preferable

Reply:
We would like not to have `expect`s or `unwrap`s that may fail, causing a panic. Infallible `unwrap`s are completely fine, e.g. if you **know** for sure that given logic will always produce `Some` or `Ok`. To my understanding, if an expression matches then you **know** for sure that the captures are not nulls, or is it not the case?

Reply:
That makes sense, fixed